### PR TITLE
MM-39171: Force RHS open when clicking from Todo list

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Uses the feature from #860 to force the RHS open when a user clicks on todo digest links

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-39171

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]
Nifty!

File: tests-e2e/cypress/integration/frontstage/rhs/open_spec.js [hunk 1]
```diff
@@ -314,6 +316,71 @@ describe('playbook run rhs', () => {
...
+                    playbookRunId: run.id,
+                    message: 'no message 1',
+                    reminder: 1,
+                });
+
+                // # Navigate to the application and a channel without a playbook run
+                cy.visit(`/${testTeam.name}/channels/off-topic`);
+
+                // # Wait until the channel loads enough to show the post textbox.
+                cy.get('#post-create').should('exist');
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/frontstage/rhs/open_spec.js:341]: Do we need to navigate to ~off-topic before navigating to the DM channel?

File: tests-e2e/cypress/integration/frontstage/rhs/open_spec.js [hunk 2]
```diff
@@ -314,6 +316,71 @@ describe('playbook run rhs', () => {
...
+                cy.executeSlashCommand('/playbook todo');
+
+                // # Open the flagged posts RHS
+                cy.get('body').then(($body) => {
+                    if ($body.find('#channelHeaderFlagButton').length > 0) {
+                        cy.get('#channelHeaderFlagButton').click({force: true});
+                    } else {
+                        cy.findByRole('button', {name: 'Select to toggle a list of saved posts.'})
+                            .click({force: true});
+                    }
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/frontstage/rhs/open_spec.js:356]: Same question as in the other PR: are there cases where we don't have the flag button?

File: tests-e2e/cypress/integration/frontstage/rhs/open_spec.js [hunk 3]
```diff
@@ -314,6 +316,71 @@ describe('playbook run rhs', () => {
...
+                // * Verify Saved Posts is open
+                cy.get('.sidebar--right__title').should('contain.text', 'Saved Posts');
+
+                // # Should show the runs overdue -- ignoring the rest
+                cy.getLastPost().within(() => {
+                    cy.get('li').then((liItems) => {
+                        expect(liItems[0]).to.contain.text(runName);
+
+                        // # Click to get to the run's channel
+                        cy.wrap(liItems[0]).get('a').contains(runName).click();
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/frontstage/rhs/open_spec.js:368]: Do we need the expect before actually clicking? The `contains` before the click will already fail if it does not exist, right?

**Review by agarciamontoro — APPROVED** [review]
Cool! A couple of non-blocking suggestions in the tests, but looks great otherwise :)

File: tests-e2e/cypress/integration/frontstage/rhs/open_spec.js (hunk 1 above)
Comment (cpoile) [reply tests-e2e/cypress/integration/frontstage/rhs/open_spec.js:341]: Ah, good catch. Thanks!

File: tests-e2e/cypress/integration/frontstage/rhs/open_spec.js (hunk 2 above)
Comment (cpoile) [reply tests-e2e/cypress/integration/frontstage/rhs/open_spec.js:356]: Yep, removing it. :)

File: tests-e2e/cypress/integration/frontstage/rhs/open_spec.js (hunk 3 above)
Comment (cpoile) [reply tests-e2e/cypress/integration/frontstage/rhs/open_spec.js:368]: yep, good point, thanks!

**Pull request merged**